tests/sparseml/pytorch/sparsification/distillation/test_per_layer_distillation.py

```diff
@@ -0,0 +1,202 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import Callable
+
+import pytest
+import torch
+from torch import Tensor
+from torch.nn import Module
+from torch.optim import Optimizer
+
+from sparseml.pytorch.sparsification import (
+    DistillationModifier,
+    Modifier,
+    ScheduledModifier,
+)
+from tests.sparseml.pytorch.helpers import LinearNet, create_optim_sgd
+from tests.sparseml.pytorch.sparsification.test_modifier import ScheduledModifierTest
+
+
+from tests.sparseml.pytorch.helpers import (  # noqa isort:skip
+    test_epoch,
+    test_loss,
+    test_steps_per_epoch,
+)
+
+
+DISTILLATION_MODIFIERS = [
+    lambda: DistillationModifier(start_epoch=0.0),
+]
+
+
+def _get_fake_batch(model_lambda):
+    batch_size = 5
+    input_shape = model_lambda.layer_descs()[0].input_size
+    return torch.randn(batch_size, *input_shape)
+
+
+@pytest.mark.skipif(
+    os.getenv("NM_ML_SKIP_PYTORCH_TESTS", False),
+    reason="Skipping pytorch tests",
+)
+@pytest.mark.parametrize("modifier_lambda", DISTILLATION_MODIFIERS, scope="function")
+@pytest.mark.parametrize("model_lambda", [LinearNet], scope="function")
+@pytest.mark.parametrize("optim_lambda", [create_optim_sgd], scope="function")
```

Comment:
we should add another parametrize for teacher_lambda, which will be different from model_lambda in some test cases